### PR TITLE
#2111 Rename entities in code that mention Menas API

### LINKED ISSUE #2111: Rename entities in code that mention Menas but REST API is meant
Menas was API+UI, but now is just UI. However, there are still plenty of variables/classes/etc. with _menas_ in the name that should be somehow renamed, for example:
- `CommonJobExecution` has _vals_: `menasBaseUrls`, `menasUrlsRetryCount`, `menasSetup`
- `MenasConnectionStringParser`
- menas-credentials.conf (and other keytab/credentials related things)
- `MenasDAO` (?)
- data-model/menas (?)

## Note
While some of them are pretty obvious renames (like `menasBaseUrls` to `restApiBaseUrls`, or menas-credentials.conf to rest_api-credentials.conf), a few are non-obvious and probably need a bit of discussion (like `MenasDAO`, whether rename it at all, and if so into what).

### PR DESCRIPTION
**Pull request by jakipatryk**
As requested in #2111, places in code/docs where Menas was mentioned (legacy from < 3.0.0, where REST API was part of Menas) but standalone Enceladus REST API was meant, were changed.

### What was changed
- Renamed `--menas-auth-keytab` to `--rest-api-auth-keytab`, and `--menas-credentials-file` to `--rest-api-auth-keytab` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/3a47196d2914b1a74728bb3b068225e06fbdf1fc) (https://github.com/AbsaOSS/enceladus/pull/2114/commits/1df097b3896867c3a20eaf14fbcd0c79f8106fde). For backwards compatibility, both old (--menas...) and new (--rest-api...) versions are available, but old one deprecated (https://github.com/AbsaOSS/enceladus/pull/2114/commits/b09fee12fac7b8fc327cab100f7cac77a5ac7c12)
- Renamed `MenasConnectionStringParser` to `RestApiConnectionStringParser` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/3a47196d2914b1a74728bb3b068225e06fbdf1fc)
- Changed `means.` prefix in `rest-api` config to `enceladus.rest.` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/e42d456d46163856819b41eb401b42229412d929) (https://github.com/AbsaOSS/enceladus/pull/2114/commits/12e03187e741b801f8cfa7931c28d17766f2652f)
- Renamed `MenasDAO` to `EnceladusDAO` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/31cc5932668125fcb3f4481b0ab7c7e70a7d0a32) (https://github.com/AbsaOSS/enceladus/pull/2114/commits/47c07901d8108c86db270fcfca8a9aea2b0f8386)
- Renamed `MenasPlugin` to `EnceladusAtumPlugin` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/a665067678140f51e41ef3e0e10a8b5a21e68388)
- Renamed `menasRestURI` to `enceladusRestURI` in Oozie config (https://github.com/AbsaOSS/enceladus/pull/2114/commits/7d62f5100ef04a44de34fb869e295ff25423bdf8)
- In `rest-api` module, renamed authentication classes/trait (https://github.com/AbsaOSS/enceladus/pull/2114/commits/4f1328ff69e5072170bfa490a4ba05f2bd268709) (https://github.com/AbsaOSS/enceladus/pull/2114/commits/08f212d2aa12b9990ed7ca8e8c61451a6c2486a1)
- Changed URI to REST API in `test_jsons` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/57d922c9b43122a171755008df5620942695e7be) and `AuthClientSuite` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/efb969b985d290a8b697a365ea8e4660c52c681f)
- Renamed `SparkMenasSchemaConvertor` to `SparkEnceladusSchemaConvertor` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/b9b1e508c5fc860eff9cd1b37118cc2038691c32)
- Renamed `za.co.absa.enceladus.model.menas` module to `za.co.absa.enceladus.model.backend` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/043c5254d7698de27fb713a4b29df0db8d9cd360)
- Renamed `MenasAttachment` to `Attachment` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/53d2572a9ccd75f47742897584802372e22a8a0d)
- Renamed `MenasReference` to `Reference` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/b04b08e8e908c3dd55f75ace1ac1a0805811e16e)
- Renamed `MenasObject` to `GenericObject` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/cc7c2256e6ee35100433e6c6749649d9abc12ffc)
- Changed `menasRef` in `AuditTrailEntry` to `ref` (https://github.com/AbsaOSS/enceladus/pull/2114/commits/b04b08e8e908c3dd55f75ace1ac1a0805811e16e#diff-971c4a452fdb16c367bf0bd7d485bcccfc6c3d2324e00104259b32bb142b9e88) (https://github.com/AbsaOSS/enceladus/pull/2114/commits/1dd5e6214e9d9740b8d6f7b8e01171182ac6938c)

Closes #2111

### CODE REVIEW

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/plugin/enceladus/EnceladusRunUrl.scala
Comment (Zejnilovic) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/common/plugin/enceladus/EnceladusRunUrl.scala:23]: I am thinking about this method. Doesn't this force us to always have the same base URL for both API and UI? 
Otherwise, this is not true, right?

**Review by Zejnilovic — COMMENTED** [review]
Seems you've missed:
- non updated properties in `examples/src/main/resources/application.conf`
- package `za.co.absa.enceladus.model.menas` should probably be `za.co.absa.enceladus.model.backend` or something, if we agreed that Menas is a UI. + there are some classes like MenasAttachment.
- file `docker.properties` needs to be updated as well
- `rest-api/src/main/scala/za/co/absa/enceladus/rest_api/utils/SchemaType.scala` line 39
- `rest-api/src/main/scala/za/co/absa/enceladus/rest_api/SpringFoxConfig.scala` lines 76 and 77 should probably say Enceladus API
- file `rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/ConfigurationController.scala` should probably speak about Enceladus environment

Otherwise looks good

**Note**: I am skipping Oozie, AFAIK that should've been deleted from Enceladus.

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/plugin/enceladus/EnceladusRunUrl.scala
Comment (jakipatryk) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/common/plugin/enceladus/EnceladusRunUrl.scala:23]: Yes it does, I was thinking if fixing this should be in this PR or another one (as it requires addition of config field to `spark-jobs` properties, not just renaming). What do you think?

**Comment by jakipatryk** [comment]
@Zejnilovic 
> file docker.properties needs to be updated as well

 Isn't this file excluded in repo? I mean it is copied from docker.properties.template on build, but not in the repo itself

**Comment by Zejnilovic** [comment]
> @Zejnilovic
> 
> > file docker.properties needs to be updated as well
> 
> Isn't this file excluded in repo? I mean it is copied from docker.properties.template on build, but not in the repo itself.
> 
> Same for examples/src/main/resources/application.conf

Spot on. My bad.

File: migrations/src/main/resources/application.properties.template
Comment (lsulak) [inline migrations/src/main/resources/application.properties.template:16]: why not `enceladus.rest_api.` as on other places? I'm just trying to understand & follow the consistent conventions. Or, the other way around, why not to have `enceladus.rest.` everywhere, instead of these `enceladus.rest_api.`?

Comment (lsulak) [reply migrations/src/main/resources/application.properties.template:16]: (just a note - also on other places, if we decide to change this)

File: rest-api/Dockerfile
Comment (lsulak) [inline rest-api/Dockerfile:43]: why not to keep it as a parameter? Does this mean that we'll have to change this string on each release?

File: rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/repositories/BaseRepositoryTest.scala
Comment (lsulak) [inline rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/repositories/BaseRepositoryTest.scala:33]: (why/) was it needed?

Comment (jakipatryk) [reply rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/repositories/BaseRepositoryTest.scala:33]: Probably I have included that one by mistake. However, with 2000 await duration integration tests are always failing on my machine, and with 8000 they work fine (on any branch). So maybe this change can stay there.

Good catch tho.

File: rest-api/Dockerfile
Comment (jakipatryk) [reply rest-api/Dockerfile:43]: You are right, my mistake.

**Review by lsulak — COMMENTED** [review]
Looks good, just a few questions above

- [X] pulled
- [x] build
- [x] ran unit tests
- [x] reviewed (quickly scanned all the changes)
- [x] ran (some spark jobs & clicked through Menas UI)

File: rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/repositories/BaseRepositoryTest.scala
Comment (lsulak) [reply rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/repositories/BaseRepositoryTest.scala:33]: if that's the case then I suggest let's keep it

File: migrations/src/main/resources/application.properties.template
Comment (jakipatryk) [reply migrations/src/main/resources/application.properties.template:16]: Not sure how it would then work in Docker's env-variables setup of these (as there the separator is `_` instead of `.`).

**Review by lsulak — APPROVED** [review]

**Review by lsulak — COMMENTED** [review]
Still LGTM, once you resolve merge conflicts I can quickly run it again and if there is nothing suspicious I think this is good to go.

**Review by lsulak — APPROVED** [review]

**Comment by lsulak** [comment]
> Looks good, just a few questions above
> 
> * [x]  pulled
> * [x]  build
> * [x]  ran unit tests
> * [x]  reviewed (quickly scanned through all the changes)
> * [x]  ran (some spark jobs & clicked through Menas UI)

I executed these steps again and it still works as expected.

**Pull request merged**
